src/main/java/org/jabref/gui/consistency/ConsistencyCheckDialog.java

```diff
@@ -0,0 +1,149 @@
+                                  BibliographyConsistencyCheck.Result result) {
+        this.libraryTab = libraryTab;
+        this.dialogService = dialogService;
+        this.preferences = preferences;
+        this.entryTypesManager = entryTypesManager;
+        this.result = result;
+
+        this.setTitle(Localization.lang("Check consistency"));
+        this.initModality(Modality.NONE);
+
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+    }
+
+    private void onSelectionChanged(ListChangeListener.Change<? extends ConsistencyMessage> change) {
+        if (change.next()) {
+            change.getAddedSubList().stream().findFirst().ifPresent(message ->
+                    libraryTab.showAndEdit(message.bibEntry()));
+        }
+    }
+
+    public ConsistencyCheckDialogViewModel getViewModel() {
+        return viewModel;
+    }
+
+    @FXML
+    public void initialize() {
+        viewModel = new ConsistencyCheckDialogViewModel(dialogService, preferences, entryTypesManager, result);
+
+        tableView.getSelectionModel().getSelectedItems().addListener(this::onSelectionChanged);
+
+        entryTypeCombo.getItems().addAll(viewModel.getEntryTypes());
+        entryTypeCombo.valueProperty().bindBidirectional(viewModel.selectedEntryTypeProperty());
+        EasyBind.listen(entryTypeCombo.getEditor().textProperty(), observable -> entryTypeCombo.commitValue());
+        entryTypeCombo.getSelectionModel().selectFirst();
+
+        FilteredList<ConsistencyMessage> filteredData = new FilteredList<>(viewModel.getTableData(), message ->
+                message.message().split("\\s+")[0].equals(viewModel.selectedEntryTypeProperty().get())
+        );
+
+        viewModel.selectedEntryTypeProperty().addListener((obs, oldValue, newValue) -> {
+            filteredData.setPredicate(message ->
+                    message.message().split("\\s+")[0].equals(newValue)
+            );
+        });
+
+        tableView.setItems(filteredData);
+
+        for (int i = 0; i < viewModel.getColumnNames().size(); i++) {
+            int columnIndex = i;
+            TableColumn<ConsistencyMessage, String> tableColumn = new TableColumn<>(viewModel.getColumnNames().get(i));
+
+            tableColumn.setCellValueFactory(row -> {
+                String[] message = row.getValue().message().split("\\s+");
+                return new ReadOnlyStringWrapper(message[columnIndex]);
+            });
+
+            tableColumn.setCellFactory(column -> new TableCell<ConsistencyMessage, String>() {
+                @Override
+                protected void updateItem(String item, boolean empty) {
+                    super.updateItem(item, empty);
+
+                    if (empty || item == null) {
+                        setGraphic(null);
+                    } else {
+                        switch (item.toLowerCase()) {
+                            case "-":
```

Comment:
Avoid magic strings. Use ConsistencyCheckDialogViewModel.REQUIRED_FIELD_AT_ENTRY_TYPE_CELL_ENTRY instead

Reply:
Symbols can be represented naturally as an enum that contains text + icon. 

Then we can write something like:

```java
Symbol.fromText(item).ifPresent(symbol -> setGraphic(symbol.icon))
                                         .orElse(() -> {
                                             setGraphic(null);
                                             setText(item);
                                         })
```